### PR TITLE
[A2-855] projects filter UI cleanup

### PR DESCRIPTION
**Pull request by srenatus**
### :nut_and_bolt: Description

Small fry that came up in ChefConf user testing. Please see the individual commit messages.

That said, I'm no UI expert, please review with that in mind 😄 

### :athletic_shoe: Demo Script / Repro Steps

- deployinate, use `automate-ui-devproxy` or `automate-ui`, use `grpcurl` to trigger different backend conditions:
- create a project:
```
[34][default:/src:0]# chef-automate dev grpcurl authz-service -- -d '{"id": "foo", "name": "foo"}'  chef.automate.domain.authz.v2.Projects/CreateProject
{
  "project": {
    "name": "foo",
    "id": "foo",
    "type": "CUSTOM",
    "projects": [
      "foo"
    ]
  }
}
```
- delete it:
```
[33][default:/src:0]# chef-automate dev grpcurl authz-service -- -d '{"id": "foo"}'  chef.automate.domain.authz.v2.Projects/DeleteProject
{

}
```

### :chains: Related Resources

- [A2-855](https://chefio.atlassian.net/browse/A2-855)

### :white_check_mark: Checklist

- [X] Necessary tests added/updated?
- [ ] Necessary docs added/updated?
- [X] Code actually executed?
- [X] Vetting performed (unit tests, lint, etc.)?

### CODE REVIEW

File: components/automate-ui/src/app/page-components/projects-filter-dropdown/projects-filter-dropdown.component.scss [hunk 1]
```diff
@@ -22,6 +22,7 @@
 
   &:disabled {
     color: $chef-primary-dark;
+    cursor: default; // no pointer when there's only one element
```
Comment (bcmdarroch) [inline components/automate-ui/src/app/page-components/projects-filter-dropdown/projects-filter-dropdown.component.scss]: found an issue here - if i've selected all projects, now i can no longer re-open the dropdown to change my project selections.

Comment (srenatus) [reply components/automate-ui/src/app/page-components/projects-filter-dropdown/projects-filter-dropdown.component.scss]: Oh really? I've tried that, I believe... let me try again.

Comment (srenatus) [reply components/automate-ui/src/app/page-components/projects-filter-dropdown/projects-filter-dropdown.component.scss]: Screenshot 2019-06-05 at 20 25 28

Having no issues opening the dropdown when I have all projects selected before (and applied that selection). Can you please elaborate?

File: components/automate-ui/src/app/page-components/projects-filter-dropdown/projects-filter-dropdown.component.ts [hunk 2]
```diff
@@ -10,7 +10,7 @@ import { ProjectsFilterOption } from 'app/services/projects-filter/projects-filt
 export class ProjectsFilterDropdownComponent implements OnChanges {
   @Input() options: ProjectsFilterOption[] = [];
 
-  @Input() selectionLabel = 'All Resources';
+  @Input() selectionLabel = 'All projects';
```
Comment (phiggins) [inline components/automate-ui/src/app/page-components/projects-filter-dropdown/projects-filter-dropdown.component.ts]: Is there a reason this doesn't use the constant?

Comment (srenatus) [reply components/automate-ui/src/app/page-components/projects-filter-dropdown/projects-filter-dropdown.component.ts]: Yeah I've found this confusing as well. I think it's a default value only that doesn't ever get used. So, I've changed it for consistency, but it might not be the best fix. Not confident enough with this to remove all the defaults... :thinking_face:

[the remark this answers is not in the document]
Comment (susanev) [reply components/automate-ui/src/app/page-components/projects-filter-dropdown/projects-filter-dropdown.component.ts]: the designs never had that `r` capitalized and i guess i missed it when that snuck in prior. @srenatus and i chatted about it already.

https://app.zeplin.io/project/5ba808703eb54372d860c737/dashboard?seid=5c464d2dd363b1bf6d2e0a4e

Comment (msorens) [reply components/automate-ui/src/app/page-components/projects-filter-dropdown/projects-filter-dropdown.component.ts]: A. It would get used if the parameter is not supplied.
B. Please change it to use the constant; just overlooked.

File: components/automate-ui/src/app/page-components/projects-filter-dropdown/projects-filter-dropdown.component.scss (hunk 1 above)
Comment (msorens) [reply components/automate-ui/src/app/page-components/projects-filter-dropdown/projects-filter-dropdown.component.scss]: I also am unable to reproduce the problem. If it is, in fact, a real issue lurking on some boundary conditions, it would not be caused by any changes from this PR.

File: components/automate-ui/src/app/page-components/projects-filter-dropdown/projects-filter-dropdown.component.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (srenatus) [reply components/automate-ui/src/app/page-components/projects-filter-dropdown/projects-filter-dropdown.component.ts]: There's a note in [the commit message](https://github.com/chef/automate/pull/504/commits/4e145dfd08156d9bd999d84890e60338ce594a73) 😉

**Comment by srenatus** [comment]
will merge when green

**Comment by srenatus** [comment]
**nope** will merge when https://github.com/chef/automate/pull/523 is in and this is rebased and green. 😓

File: components/automate-ui/src/app/page-components/projects-filter-dropdown/projects-filter-dropdown.component.scss (hunk 1 above)
Comment (bcmdarroch) [reply components/automate-ui/src/app/page-components/projects-filter-dropdown/projects-filter-dropdown.component.scss]: Oops this was due to an old gateway build running on my machine. nevermind! 😅
